Jenkinsfile

```diff
@@ -188,12 +134,59 @@ pipeline {
                       }
                       unstash 'source'
                       dir("${BASE_DIR}"){
-                        bat """
-                        nuget restore ElasticApmAgent.sln
-                        msbuild
-                        """
+                        bat '.ci/windows/msbuild.bat'
+                      }
+                    }
+                  }
+                  /**
+                  Execute unit tests.
+                  */
+                  stage('Test') {
+                    steps {
+                      dir("${BASE_DIR}"){
+                        deleteDir()
+                      }
+                      unstash 'source'
+                      dir("${BASE_DIR}"){
+                        powershell label: 'Install test tools', script: "${readFile('./.ci/windows/test-tools.ps1')}"
+                        bat label: 'Build', script: '.ci/windows/msbuild.bat'
+                        bat label: 'Test & coverage', script: '.ci/windows/test.bat'
+                        powershell label: 'Convert Test Results to junit format', script: "${readFile('.ci/windows/convert.ps1')}"
+                      }
+                    }
+                    post {
+                      always {
+                        junit(allowEmptyResults: false,
+                          keepLongStdio: true,
+                          testResults: "${BASE_DIR}/**/junit-*.xml,${BASE_DIR}/target/**/TEST-*.xml")
+                      }
+                    }
+                  }
+                }
+              }
+              stage('Windows .NET Core'){
+                agent { label 'windows-2016' }
+                options { skipDefaultCheckout() }
+                environment {
+                  HOME = "${env.WORKSPACE}"
+                  DOTNET_ROOT = "${env.WORKSPACE}\\dotnet"
+                  VS_HOME = "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Enterprise"
+                  MSBuildSDKsPath = "${env.DOTNET_ROOT}\\sdk\\2.1.505\\Sdks"
+                  PATH = "${env.PATH};${env.HOME}\\bin;${env.DOTNET_ROOT};${env.DOTNET_ROOT}\\tools;\"${env.VS_HOME}\\MSBuild\\15.0\\Bin\""
+                }
+                stages{
+                  /**
+                  Checkout the code and stash it, to use it on other stages.
+                  */
+                  stage('Install tools') {
+                    steps {
+                      deleteDir()
+                      unstash 'source'
+                      dir("${HOME}"){
+                        powershell label: 'Install tools', script: """${readFile("${BASE_DIR}/.ci/windows/tools.ps1")}"""
```

Comment:
```suggestion
                        powershell label: 'Install tools', script: '.\\.ci\\windows\\tools.ps1'
```

Reply:
In order to install the tools it's required to be within the HOME folder and therefore it's required to use the BASE_DIR to be able to point out to the right folder where the script is stored.

Reply:
ok